Race in `TestAwaitUtilCancelTx` caused by the test design

`TestAwaitUtilCancelTx` is failing sometimes because of the fact that original transaction is being accepted instead of conflicting one. It happens because chain executor's chain continiousely produces blocks, so occasionally the block is being accepted right after main transaction submission. Go 1.22, macOS, firstly discovered at https://github.com/nspcc-dev/neo-go/actions/runs/8328012697/job/22787046121?pr=3357.

The possible solution is to add two branches into this test. One branch should handle the case of original transaction acceptance, the other branch should handle the conflicting transaction acceptance. This situation is very natural, so let the test follow both cases (similar to the way how `TestWalletClaimGas` works.
```
2024-03-18T14:12:38.1281770Z === RUN   TestAwaitUtilCancelTx
2024-03-18T14:12:38.1284550Z     logger.go:146: 2024-03-18T14:11:11.985Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2024-03-18T14:12:38.1289250Z     logger.go:146: 2024-03-18T14:11:11.985Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2024-03-18T14:12:38.1293660Z     logger.go:146: 2024-03-18T14:11:11.985Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2024-03-18T14:12:38.1297780Z     logger.go:146: 2024-03-18T14:11:11.985Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2024-03-18T14:12:38.1302150Z     logger.go:146: 2024-03-18T14:11:11.985Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2024-03-18T14:12:38.1306370Z     logger.go:146: 2024-03-18T14:11:11.986Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 864000}
2024-03-18T14:12:38.1310270Z     logger.go:146: 2024-03-18T14:11:11.986Z	INFO	no storage version found! creating genesis block
2024-03-18T14:12:38.1312860Z     logger.go:146: 2024-03-18T14:11:12.032Z	INFO	MaxIteratorResultItems is not set or wrong, setting default value	{"MaxIteratorResultItems": 100}
2024-03-18T14:12:38.1317080Z     logger.go:146: 2024-03-18T14:11:12.033Z	INFO	MaxFindResultItems is not set or wrong, setting default value	{"MaxFindResultItems": 100}
2024-03-18T14:12:38.1321560Z     logger.go:146: 2024-03-18T14:11:12.033Z	INFO	MaxFindStorageResultItems is not set or wrong, setting default value	{"MaxFindStorageResultItems": 50}
2024-03-18T14:12:38.1325620Z     logger.go:146: 2024-03-18T14:11:12.033Z	INFO	MaxNEP11Tokens is not set or wrong, setting default value	{"MaxNEP11Tokens": 100}
2024-03-18T14:12:38.1329240Z     logger.go:146: 2024-03-18T14:11:12.033Z	INFO	MaxRequestBodyBytes is not set or wong, setting default value	{"MaxRequestBodyBytes": 5242880}
2024-03-18T14:12:38.1332860Z     logger.go:146: 2024-03-18T14:11:12.033Z	DEBUG	done processing headers	{"headerIndex": 1, "blockHeight": 0, "took": "634.803µs"}
2024-03-18T14:12:38.1335730Z     logger.go:146: 2024-03-18T14:11:12.033Z	INFO	MaxRequestHeaderBytes is not set or wong, setting default value	{"MaxRequestHeaderBytes": 1048576}
2024-03-18T14:12:38.1338540Z     logger.go:146: 2024-03-18T14:11:12.034Z	INFO	MaxWebSocketClients is not set or wrong, setting default value	{"MaxWebSocketClients": 64}
2024-03-18T14:12:38.1341380Z     logger.go:146: 2024-03-18T14:11:12.034Z	INFO	starting rpc-server	{"endpoint": "127.0.0.1:0"}
2024-03-18T14:12:38.1344790Z     logger.go:146: 2024-03-18T14:11:12.040Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-18T14:12:38.1354560Z     logger.go:146: 2024-03-18T14:11:12.042Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2024-03-18T14:12:38.1356870Z     logger.go:146: 2024-03-18T14:11:12.081Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-18T14:12:38.1361330Z     logger.go:146: 2024-03-18T14:11:12.083Z	DEBUG	processing rpc request	{"method": "getnep17balances", "params": "[56c989e76f9a2ca05bb5caa6c96f524d905accd8]"}
2024-03-18T14:12:38.1373300Z     logger.go:146: 2024-03-18T14:11:12.087Z	DEBUG	processing rpc request	{"method": "invokescript", "params": "[CxEMFNwk6EPkEi4hj1BzCuuZPYVDDVeBDBTYzFqQTVJvyabKtVugLJpv54nJVhTAHwwIdHJhbnNmZXIMFPVj6kC8KD1NDgXEjqMFs/Kgc0DvQWJ9W1I5 ]"}
2024-03-18T14:12:38.1376930Z     logger.go:146: 2024-03-18T14:11:12.091Z	DEBUG	processing rpc request	{"method": "getblockcount", "params": "[]"}
2024-03-18T14:12:38.1384980Z     logger.go:146: 2024-03-18T14:11:12.092Z	DEBUG	processing rpc request	{"method": "calculatenetworkfee", "params": "[AL+xcc+0P5gAAAAAAAAAAAAAAAAABAAAAAHYzFqQTVJvyabKtVugLJpv54nJVgEAVwsRDBTcJOhD5BIuIY9QcwrrmT2FQw1XgQwU2MxakE1Sb8mmyrVboCyab+eJyVYUwB8MCHRyYW5zZmVyDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQEAKhEMIQKzYiv0AXvf4xfFiu1fTHU/IGt9uJYEb6fXdLvEv3+NwhFBntDcOg==]"}
2024-03-18T14:12:38.1400300Z     logger.go:146: 2024-03-18T14:11:12.093Z	DEBUG	processing rpc request	{"method": "sendrawtransaction", "params": "[AL+xcc+0P5gAAAAAAPTGEgAAAAAABAAAAAHYzFqQTVJvyabKtVugLJpv54nJVgEAVwsRDBTcJOhD5BIuIY9QcwrrmT2FQw1XgQwU2MxakE1Sb8mmyrVboCyab+eJyVYUwB8MCHRyYW5zZmVyDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQFCDEA51mOBl0nGji0ZR/T1FTGuRIeBphZ6hut5HtxYtGBhkA55np/tFBSpeLAJdL0zKwtUualypxdPDnaxVBL3/m/IKhEMIQKzYiv0AXvf4xfFiu1fTHU/IGt9uJYEb6fXdLvEv3+NwhFBntDcOg==]"}
2024-03-18T14:12:38.1408640Z     logger.go:146: 2024-03-18T14:11:12.097Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-18T14:12:38.1411190Z     logger.go:146: 2024-03-18T14:11:12.098Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2024-03-18T14:12:38.1413360Z     logger.go:146: 2024-03-18T14:11:12.114Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-18T14:12:38.1417410Z     logger.go:146: 2024-03-18T14:11:12.118Z	DEBUG	processing rpc request	{"method": "getrawtransaction", "params": "[b1b2ba3bc8fbb340cb12841306fc54b8c8fe4477a98956b02fb8f447c578a46b 1]"}
2024-03-18T14:12:38.1427580Z     logger.go:146: 2024-03-18T14:11:12.135Z	DEBUG	done processing headers	{"headerIndex": 2, "blockHeight": 1, "took": "242.594µs"}
2024-03-18T14:12:38.1431180Z     logger.go:146: 2024-03-18T14:11:12.163Z	DEBUG	processing rpc request	{"method": "invokescript", "params": "[QA== ]"}
2024-03-18T14:12:38.1433370Z     logger.go:146: 2024-03-18T14:11:12.167Z	DEBUG	processing rpc request	{"method": "getblockcount", "params": "[]"}
2024-03-18T14:12:38.1439410Z     logger.go:146: 2024-03-18T14:11:12.171Z	DEBUG	processing rpc request	{"method": "calculatenetworkfee", "params": "[AEK492AAAAAAAAAAAAAAAAAAAAAABQAAAAHYzFqQTVJvyabKtVugLJpv54nJVgEBIWukeMVH9LgvsFaJqXdE/si4VPwGE4QSy0Cz+8g7urKxAUABACoRDCECs2Ir9AF73+MXxYrtX0x1PyBrfbiWBG+n13S7xL9/jcIRQZ7Q3Do=]"}
2024-03-18T14:12:38.1447070Z     logger.go:146: 2024-03-18T14:11:12.172Z	DEBUG	processing rpc request	{"method": "sendrawtransaction", "params": "[AEK492AAAAAAAAAAAPXGEgAAAAAABAAAAAHYzFqQTVJvyabKtVugLJpv54nJVgEBIWukeMVH9LgvsFaJqXdE/si4VPwGE4QSy0Cz+8g7urKxAUABQgxAKMFykB4BdUMOcwyVKq8OpiTLkCxy0miTzoJZSpWghqf04VXVHNBS0RYZ35suMn4plDa7N8Ky7Dpq5C78J29KPCoRDCECs2Ir9AF73+MXxYrtX0x1PyBrfbiWBG+n13S7xL9/jcIRQZ7Q3Do=]"}
2024-03-18T14:12:38.1461860Z     logger.go:146: 2024-03-18T14:11:12.173Z	INFO	Error encountered with rpc request	{"code": -507, "cause": "invalid attribute: conflicting transaction b1b2ba3bc8fbb340cb12841306fc54b8c8fe4477a98956b02fb8f447c578a46b is already on chain", "method": "sendrawtransaction", "params": "[AEK492AAAAAAAAAAAPXGEgAAAAAABAAAAAHYzFqQTVJvyabKtVugLJpv54nJVgEBIWukeMVH9LgvsFaJqXdE/si4VPwGE4QSy0Cz+8g7urKxAUABQgxAKMFykB4BdUMOcwyVKq8OpiTLkCxy0miTzoJZSpWghqf04VXVHNBS0RYZ35suMn4plDa7N8Ky7Dpq5C78J29KPCoRDCECs2Ir9AF73+MXxYrtX0x1PyBrfbiWBG+n13S7xL9/jcIRQZ7Q3Do=]"}
2024-03-18T14:12:38.1477430Z failed to send conflicting transaction: Invalid transaction attribute (-507) - invalid attribute: conflicting transaction b1b2ba3bc8fbb340cb12841306fc54b8c8fe4477a98956b02fb8f447c578a46b is already on chain
2024-03-18T14:12:38.1482300Z     executor.go:286: 
2024-03-18T14:12:38.1484120Z         	Error Trace:	/Users/runner/work/neo-go/neo-go/internal/testcli/executor.go:286
2024-03-18T14:12:38.1487030Z         	            				/Users/runner/work/neo-go/neo-go/cli/util/util_test.go:171
2024-03-18T14:12:38.1488560Z         	Error:      	Received unexpected error:
2024-03-18T14:12:38.1494180Z         	            	failed to send conflicting transaction: Invalid transaction attribute (-507) - invalid attribute: conflicting transaction b1b2ba3bc8fbb340cb12841306fc54b8c8fe4477a98956b02fb8f447c578a46b is already on chain
2024-03-18T14:12:38.1500160Z         	Test:       	TestAwaitUtilCancelTx
2024-03-18T14:12:38.1502210Z     logger.go:146: 2024-03-18T14:11:12.196Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:49239"}
2024-03-18T14:12:38.1505990Z     logger.go:146: 2024-03-18T14:11:12.197Z	INFO	persisted to disk	{"blocks": 2, "keys": 166, "headerHeight": 2, "blockHeight": 2, "took": "121.248µs"}
2024-03-18T14:12:38.1508430Z --- FAIL: TestAwaitUtilCancelTx (0.21s)
2024-03-18T14:12:38.1509280Z FAIL
2024-03-18T14:12:38.1510140Z FAIL	github.com/nspcc-dev/neo-go/cli/util	0.463s
```

--- a/cli/util/util_test.go
+++ b/cli/util/util_test.go
@@ -3,6 +3,7 @@ package util_test
 import (
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 	"time"
 
@@ -169,11 +170,22 @@ func TestAwaitUtilCancelTx(t *testing.T) {
 
 	e.In.WriteString("one\r")
 	e.Run(t, append(args, txHash.StringLE())...)
-	e.CheckNextLine(t, "Conflicting transaction accepted")
-	resHash, _ := e.CheckAwaitableTxPersisted(t)
 
-	require.Eventually(t, func() bool {
-		_, aerErr := e.Chain.GetAppExecResults(resHash.Hash(), trigger.Application)
-		return aerErr == nil
-	}, time.Second*2, time.Millisecond*50)
+	response := e.GetNextLine(t)
+	if strings.Contains(response, "Conflicting transaction accepted") {
+		resHash, _ := e.CheckAwaitableTxPersisted(t)
+		require.Eventually(t, func() bool {
+			_, aerErr := e.Chain.GetAppExecResults(resHash.Hash(), trigger.Application)
+			return aerErr == nil
+		}, time.Second*2, time.Millisecond*50)
+	} else if strings.Contains(response, "Target transaction accepted") {
+		require.Eventually(t, func() bool {
+			_, _, err := e.Chain.GetTransaction(txHash)
+			require.NoError(t, err, "original transaction should be on chain")
+			_, aerErr := e.Chain.GetAppExecResults(txHash, trigger.Application)
+			return aerErr == nil
+		}, time.Second*2, time.Millisecond*50)
+	} else {
+		t.Fatalf("unexpected response: %s", response)
+	}
 }

